tests/Doctrine/Tests/ORM/Query/QueryTest.php

```diff
@@ -183,6 +186,24 @@ public function testProcessParameterValueClassMetadata()
         );
     }
 
+    /**
+     * @group 5936
+     */
+    public function testProcessParameterValueEmbeddable()
+    {
+        $query  = $this->_em->createQuery("SELECT a FROM " . DDC3597Image::class . " a");
+
+        // load class meta data
+        $this->_em->getClassMetadata(DDC3597Image::class);
+        $this->assertTrue($this->_em->getMetadataFactory()->hasMetadataFor(DDC3597Image::class));
+        $this->assertTrue($this->_em->getMetadataFactory()->hasMetadataFor(DDC3597Dimension::class));
+
+        $this->assertInstanceOf(
+            DDC3597Dimension::class,
+            $query->processParameterValue(new DDC3597Dimension())
+        );
```

Comment:
I based the test on this one: https://github.com/doctrine/doctrine2/blob/master/tests/Doctrine/Tests/ORM/Query/QueryTest.php#L177